hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/Client.java

```diff
@@ -1420,6 +1350,7 @@ public void stop() {
     // wake up all connections
     for (Connection conn : connections.values()) {
       conn.interrupt();
+      conn.rpcRequestThread.interrupt();
```

Comment:
Why don't we use `conn.close()` first? Because it might be blocked by `IOUtils.closeStream(ipcStreams)` or `socket.close()`?

Reply:
Given how touchy this code is, I really don't want to make any unrelated changes.